src/mantine-hooks/src/use-viewport-size/use-viewport-size.ts

```diff
@@ -0,0 +1,36 @@
+import { useCallback, useState, useEffect } from 'react';
+import { useWindowEvent } from '../use-window-event/use-window-event';
+
+const eventListerOptions = {
+  passive: true,
+};
+
+export function useViewportSize(): Readonly<{
```

Comment:
Return type here is also redundant, TypeScript will handle return values correctly, there is no need to make them readonly